price-estimator/src/filter.rs

```diff
@@ -213,17 +203,19 @@ async fn estimate_buy_amount(
         let amount = Amount::BaseUnits(sell_amount_in_quote);
         (amount, amount.as_atoms(&token_info) as _)
     };
-    let transitive_order = orderbook
-        .pricegraph(query.batch_id, PricegraphKind::Raw)
+    let rounding_buffer = orderbook.rounding_buffer(token_pair).await;
+    let pricegraph = orderbook
+        .pricegraph(query.batch_id, PricegraphKind::WithRoundingBuffer)
         .await
-        .map_err(error::internal_server_rejection)?
-        .order_for_sell_amount(token_pair, sell_amount_in_quote_atoms);
-
-    let mut buy_amount_in_base = Amount::Atoms(
-        transitive_order
-            .map(|order| apply_rounding_buffer(order.buy, price_rounding_buffer))
-            .unwrap_or_default() as _,
+        .map_err(error::internal_server_rejection)?;
+    // This reduced sell amount is what the solver would see after applying the rounding buffer.
+    let transitive_order = pricegraph.order_for_sell_amount(
+        token_pair,
+        f64::max(sell_amount_in_quote_atoms - rounding_buffer, 0.0),
```

Comment:
I think this might be backwards:
```suggestion
        f64::max(sell_amount_in_quote_atoms + rounding_buffer, 0.0),
```

Since we want to estimate an order such that when placed, it has the amount specified by the parameter correct? I think if we reduce the sell amount here, we will get "better" prices, which is the opposite of what we want as far as I can tell.

Reply:
Tested:
With minus: 1.010709
With plus   : 1.011105
So the user will get more buy amount with plus than with minus, so minus is more conservative which is what we want. The question that the user wants answered is what they should put as the buy amount and still get matched. Internally the solver will see their order as having slightly less sell amount than they have set so the expected buy amount is lower too. This seems inline with the test, or am I switching something up?

Reply:
Oh duh... The price might be better but the total amount will not be. I think you are correct and we need to minus. Sorry for the confusion.